src/main/java/gov/cabinetoffice/gap/applybackend/service/GrantMandatoryQuestionService.java

```diff
@@ -96,13 +89,39 @@ public GrantMandatoryQuestions createMandatoryQuestion(GrantScheme scheme, Grant
     }
 
 
-    public GrantMandatoryQuestions updateMandatoryQuestion(GrantMandatoryQuestions grantMandatoryQuestions) {
+    public GrantMandatoryQuestions updateMandatoryQuestion(GrantMandatoryQuestions grantMandatoryQuestions, GrantApplicant grantApplicant) {
+        if(grantMandatoryQuestions.getStatus().equals(GrantMandatoryQuestionStatus.COMPLETED)) {
+            final Submission submission = grantMandatoryQuestions.getSubmission();
+            grantMandatoryQuestions.setGapId(submission == null ? generateGapId(grantApplicant.getId()) : submission.getGapId());
```

Comment:
It's personal preference but I really really dislike passing ternary operations like this into method params. Can we set a value using the ternary then pass it into the `setGapId` method after it's been set?